sdk/examples/wallet/spammer.rs

```diff
@@ -0,0 +1,229 @@
+//!
+//! Rename `.env.example` to `.env` first, then run the command:
+//! ```sh
+//! cargo run --release --features=wallet --example spammer
+//! ```
+
+use std::{env::var, time::Duration};
+
+use iota_sdk::{
+    client::{
+        constants::SHIMMER_COIN_TYPE,
+        request_funds_from_faucet,
+        secret::{mnemonic::MnemonicSecretManager, SecretManager},
+    },
+    types::block::{address::Bech32Address, payload::transaction::TransactionId},
+    wallet::{account::FilterOptions, Account, ClientOptions, Result, SendAmountParams, Wallet},
+};
+use tokio::{
+    task::JoinSet,
+    time::{sleep, Instant},
+};
+
+// The account alias used in this example.
+const ACCOUNT_ALIAS: &str = "spammer";
+// The number of spamming rounds.
+const NUM_ROUNDS: usize = 1000;
+// The amount to send in each transaction
+const SEND_AMOUNT: u64 = 1_000_000;
+// The number of simultaneous transactions
+const NUM_SIMULTANEOUS_TXS: usize = 64;
+
+#[tokio::main(flavor = "multi_thread")]
+async fn main() -> Result<()> {
+    // This example uses secrets in environment variables for simplicity which should not be done in production.
+    dotenvy::dotenv().ok();
+
+    let num_simultaneous_txs = NUM_SIMULTANEOUS_TXS.min(num_cpus::get());
+
+    println!("Spammer set up to issue {num_simultaneous_txs} transactions simultaneously.");
+
+    // Restore wallet from a mnemonic phrase.
+    let client_options = ClientOptions::new().with_node(&std::env::var("NODE_URL").unwrap())?;
+    let secret_manager =
+        MnemonicSecretManager::try_from_mnemonic(&std::env::var("NON_SECURE_USE_OF_DEVELOPMENT_MNEMONIC_1").unwrap())?;
+    let wallet = Wallet::builder()
+        .with_secret_manager(SecretManager::Mnemonic(secret_manager))
+        // .with_storage_path(&var("WALLET_DB_PATH").unwrap())
+        .with_client_options(client_options)
+        .with_coin_type(SHIMMER_COIN_TYPE)
+        .finish()
+        .await?;
+    let account = get_or_create_account(&wallet, ACCOUNT_ALIAS).await?;
+
+    let recv_address = *account.addresses().await?[0].address();
+
+    // Ensure there are enough available funds for spamming.
+    let available_funds = ensure_enough_funds(&account, &recv_address, num_simultaneous_txs).await?;
+    account.sync(None).await?;
+
+    let split_amount = available_funds / 2;
+    let output_amount = split_amount / 127u64;
+
+    println!("Consolidating outputs...");
+    if let Ok(transaction) = account.consolidate_outputs(false, Some(127)).await {
+        wait_for_inclusion(&transaction.transaction_id, &account).await?;
+
+        account.sync(None).await?;
+    } else {
+        println!("not necessary");
+    }
+
+    let num_unspent_basic_outputs = account
+        .unspent_outputs(FilterOptions {
+            output_types: Some(vec![3]),
+            ..Default::default()
+        })
+        .await?
+        .len();
+
+    if num_unspent_basic_outputs < 127 {
```

Comment:
shouldn't this be >?

Reply:
So the idea (actually Thoralf's idea) was to maximally split the funds into outputs. The 128th output will contain the remainder amount. This way the ISA has many unspent outputs to pick for assembling the transferred amount and we can create transactions in parallel where all have their unique unspent inputs. This line ensures that there are always 127 unspent outputs the ISA can choose from.